### PR TITLE
Add devfile for the Vert.x Java Stack with runtime ubi8/openjdk-11 image 

### PR DESCRIPTION
**Pull request by RomanNikitenko**
### What does this PR do?:
At the moment `quay.io/eclipse/che-java11-maven:7.37.2` is used as a runtime image for the Vert.x Java Stack.
The image was not updated about 2 years: https://quay.io/repository/eclipse/che-java11-maven?tab=tags.
Also, it's not possible to start Che-Code using that image - starting a workspace fails with the error: `/checode/checode-linux-libc/node: error while loading shared libraries: libbrotlidec.so.1: cannot open shared object file: No such file or directory`.

I propose to add a new version of the devfile with the runtime image `registry.access.redhat.com/ubi8/openjdk-11:1.18-2`. 

### Which issue(s) this PR fixes:
https://github.com/eclipse/che/issues/22785
https://github.com/eclipse/che/issues/22525

### PR acceptance criteria:

- [ ] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [ ] Test automation
_Does this repository's tests pass with your changes?_
- [ ] Documentation
_Does any documentation need to be updated with your changes?_
- [ ] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### How to test changes / Special notes to the reviewer:

### CODE REVIEW

**Review by thepetk — COMMENTED** [review]
I think a better approach could be here to create a new version of the vert.x stack. We can take as an example the [java-wildfly](https://github.com/devfile/registry/tree/main/stacks/java-wildfly) stack. WDYT?

**Review by ibuziuk — APPROVED** [review]

**Review by michael-valdron — COMMENTED** [review]
> I think a better approach could be here to create a new version of the vert.x stack. We can take as an example the [java-wildfly](https://github.com/devfile/registry/tree/main/stacks/java-wildfly) stack. WDYT?

+1

**Comment by RomanNikitenko** [comment]
> I think a better approach could be here to create a new version of the vert.x stack. We can take as an example the [java-wildfly](https://github.com/devfile/registry/tree/main/stacks/java-wildfly) stack. WDYT?

thank you for the quick feedback!

I'm going to provide the corresponding changes, but first I would like to confirm that I understand correctly what I should to do:
- create folder `1.2.0` and move the current devfile without my changes to that folder ([it has 1.2.0 version](https://github.com/devfile/registry/blob/main/stacks/java-vertx/devfile.yaml#L12))
- create folder `1.3.0` and place the devfile with my changes in that folder?

thanks in advance!

**Comment by michael-valdron** [comment]
> I'm going to provide the corresponding changes, but first I would like to confirm that I understand correctly what I should to do:
> 
>  - create folder 1.2.0 and move the current devfile without my changes to that folder ([it has 1.2.0 version](https://github.com/devfile/registry/blob/main/stacks/java-vertx/devfile.yaml#L12))
>  - create folder 1.3.0 and place the devfile with my changes in that folder?
>  - provide stack.yaml, 1.3.0 is default?

@RomanNikitenko These are the correct changes :+1: 

Additionally, with the versions defined in their devfiles as [1.2.0 version](https://github.com/devfile/registry/blob/main/stacks/java-vertx/devfile.yaml#L12) currently does.

**Comment by thepetk** [comment]
> > I think a better approach could be here to create a new version of the vert.x stack. We can take as an example the [java-wildfly](https://github.com/devfile/registry/tree/main/stacks/java-wildfly) stack. WDYT?
> 
> thank you for the quick feedback!
> 
> I'm going to provide the corresponding changes, but first I would like to confirm that I understand correctly what I should to do:
> 
> * create folder `1.2.0` and move the current devfile without my changes to that folder ([it has 1.2.0 version](https://github.com/devfile/registry/blob/main/stacks/java-vertx/devfile.yaml#L12))
> * create folder `1.3.0` and place the devfile with my changes in that folder?
> * provide `stack.yaml`, `1.3.0` is default?
> 
> thanks in advance!

Yes I think this is correct!

**Review by michael-valdron — CHANGES_REQUESTED** [review]
One correction needed:

```diff
- stacks/java-vertx/1.3..0/devfile.yaml
+ stacks/java-vertx/1.3.0/devfile.yaml
```

**Review by thepetk — APPROVED** [review]
/lgtm

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Comment by ibuziuk** [comment]
@thepetk @michael-valdron folks, could you please clarify who can provide guidance regarding the odo v3 failing PR check?

**Comment by RomanNikitenko** [comment]
I've changed default version for the Vert.x Java stack.
Maybe 1.3.0 should be used [here](https://github.com/devfile/registry/blob/69e1247f1dd5f6ab8985fe0b1f37c9a9cd3dbeac/tests/check_odov3.sh#L9-L22)

**Comment by michael-valdron** [comment]
>> @thepetk @michael-valdron folks, could you please clarify who can provide guidance regarding the odo v3 failing PR check?

> I've changed default version for the Vert.x Java stack.
> Maybe 1.3.0 should be used [here](https://github.com/devfile/registry/blob/69e1247f1dd5f6ab8985fe0b1f37c9a9cd3dbeac/tests/check_odov3.sh#L9-L22)

@RomanNikitenko is right, we skip vertx stack starter projects in the odo checks because they exceed the resource limit of GitHub workflows, following are some past related PRs:
- https://github.com/devfile/registry/pull/132
- https://github.com/devfile/registry/pull/264
- https://github.com/devfile/registry/pull/280

File: tests/check_odov3.sh [hunk 1]
```diff
@@ -6,20 +6,20 @@ stackDirs=$(bash "$(pwd)/tests/get_stacks.sh")
...
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-crud-example-redhat" \
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-crud-example" \
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-configmap-example" \
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-http-example-redhat" \
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-secured-http-example" \
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-istio-circuit-breaker-booster" \
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-istio-distributed-tracing-booster" \
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-istio-routing-booster" \
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-istio-security-booster" \
-  --skip="stack: java-vertx version: 1.2.0 starter: vertx-messaging-work-queue-booster" \
```
Comment (michael-valdron) [inline tests/check_odov3.sh:22]: We'll need to keep both versions here, mainly so that this PR can pass since `1.2.0` was refactored in these changes.

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Comment by RomanNikitenko** [comment]
thank you all for your support and guidance to get the PR merged!

**Pull request merged**